File: src/main/java/com/fish/entity/pojo/Like.java
package com.fish.entity.pojo;


import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_like")
public class Like implements Serializable {
    @Id(keyType = KeyType.Auto)
    private Long likeId;
    /**
     * 题解id
     */
    private Long solutionId;
    /**
     * 评论id
     */
    private Long commentId;
    /**
     * 点赞的用户
     */
    private String userId;
}

File: src/main/kotlin/com/fish/entity/pojo/Like.kt
package com.fish.entity.pojo

import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_like")
open class Like(
    likeId: Long?,
    solutionId: Long?,
    commentId: Long?,
    userId: String?
) : Serializable {
    @Id(keyType = KeyType.Auto)
    val likeId: Long? = null
    /**
     * 题解id
     */
    val solutionId: Long? = null
    /**
     * 评论id
     */
    val commentId: Long? = null
    /**
     * 点赞的用户
     */
    val userId: String? = null
}
